sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -336,10 +336,12 @@ def __init__(self, options):
     # for GCS staging locations where the potential for such clashes is high.
     if self.google_cloud_options.staging_location.startswith('gs://'):
       path_suffix = '%s.%f' % (self.google_cloud_options.job_name, time.time())
-      self.google_cloud_options.staging_location = utils.path.join(
+      filesystem = get_filesystem(self.google_cloud_options.staging_location)
+      self.google_cloud_options.staging_location = filesystem.join(
           self.google_cloud_options.staging_location, path_suffix)
-      self.google_cloud_options.temp_location = utils.path.join(
+      self.google_cloud_options.temp_location = filesystem.join(
```

Comment:
Yes, I think me and Cham discussed it at some point. I can work on that in another PR and clean the get_filesystem infiltration everywhere.